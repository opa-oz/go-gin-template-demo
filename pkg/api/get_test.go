package api_test

import (
	"github.com/gkampitakis/go-snaps/snaps"
	"net/http"
	"net/http/httptest"
	"server-template/pkg/api"
	"testing"

	"github.com/gin-gonic/gin"
	"github.com/go-playground/assert/v2"
)

func TestGetRoute(t *testing.T) {
	r := gin.Default()

	r.GET("/get/:name", api.Get)
	r.GET("/g/:name", api.Get)

	t.Run("if keys is not set, get returns 0", func(t *testing.T) {
		w := httptest.NewRecorder()
		req, _ := http.NewRequest(http.MethodGet, "/get/test?index=0", nil)
		r.ServeHTTP(w, req)

		snaps.MatchSnapshot(t, w.Body.String())
		assert.Equal(t, http.StatusOK, w.Code)
	})
}
