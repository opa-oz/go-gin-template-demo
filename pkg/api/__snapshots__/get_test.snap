
[TestGetRoute/if_keys_is_not_set,_get_returns_0 - 1]
{"name":"test","value":0}
---
